Settings.c

```diff
@@ -294,32 +294,35 @@ static void writeFields(FILE* fd, const ProcessField* fields, Hashtable* columns
    fprintf(fd, "\n");
 }
 
-static void writeMeters(const Settings* this, FILE* fd, int column) {
+static void writeMeters(const Settings* this, FILE* fd, char separator, int column) {
    const char* sep = "";
    for (int i = 0; i < this->columns[column].len; i++) {
       fprintf(fd, "%s%s", sep, this->columns[column].names[i]);
       sep = " ";
    }
-   fprintf(fd, "\n");
+   fputc(separator, fd);
 }
 
-static void writeMeterModes(const Settings* this, FILE* fd, int column) {
+static void writeMeterModes(const Settings* this, FILE* fd, char separator, int column) {
    const char* sep = "";
    for (int i = 0; i < this->columns[column].len; i++) {
       fprintf(fd, "%s%d", sep, this->columns[column].modes[i]);
       sep = " ";
    }
-   fprintf(fd, "\n");
+   fputc(separator, fd);
 }
 
 int Settings_write(const Settings* this, bool onCrash) {
    FILE* fd;
+   char separator;
    if (onCrash) {
       fd = stderr;
+      separator = '%';
```

Comment:
```suggestion
      separator = '|';
```
More logical as a line separator than %.

Reply:
We could even use `;` or `,` given none of our settings contain those characters …

Reply:
Switched to `;` as separator.